### PR TITLE
Add & delete user favorites route

### PR DESCRIPTION
**Pull request by janinebar**

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -32,6 +32,7 @@ func main() {
 	router.HandleFunc("/profile", routes.GetProfileHandler).Methods("GET")
 	router.HandleFunc("/foodtrucks", routes.PostFoodTrucksHandler).Methods("POST")
 	router.HandleFunc("/reviews", routes.PostReviewsHandler).Methods("POST")
+	router.HandleFunc("/user/favorite/{foodTruckID", routes.PutFavoriteHandler).Methods("PUT")
```
Comment (kftang) [inline server.go]: missing }

File: routes/users.go [hunk 2]
```diff
@@ -152,6 +153,51 @@ func PostLoginHandler(w http.ResponseWriter, r *http.Request) {
...
+	if !userLoggedIn {
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+
+	var updateOperator string
+
+	// Determine if an add or delete
+	action := r.URL.Query().Get("action")
+	if action == "add" {
```
Comment (kftang) [inline routes/users.go:179]: Mmm, I'd say a different route for add and deleting favorites is favorable here, query parameters are really just for GET routes

File: routes/users.go [hunk 3]
```diff
@@ -152,6 +153,51 @@ func PostLoginHandler(w http.ResponseWriter, r *http.Request) {
...
+	} else {
+		log.Printf("ERROR: Incorrect action to edit user favorites.")
+		w.WriteHeader(http.StatusBadRequest)
+	}
+
+	updateFavesFilter := bson.M{updateOperator: bson.M{"favorites": foodTruckID}}
+
+	_, err := Db.Collection("users").UpdateOne(r.Context(), queries.WithID(userID), updateFavesFilter)
+	if err != nil {
+		log.Fatal(err)
```
Comment (kftang) [inline routes/users.go]: Printf("ERROR: %v", err)

we don't want an error to crash the server

File: routes/users.go [hunk 4]
```diff
@@ -152,6 +153,51 @@ func PostLoginHandler(w http.ResponseWriter, r *http.Request) {
...
+		w.WriteHeader(http.StatusNotFound)
```
Comment (kftang) [inline routes/users.go:194]: need to return or else there will be a 200

File: routes/users.go [hunk 5]
```diff
@@ -152,6 +153,51 @@ func PostLoginHandler(w http.ResponseWriter, r *http.Request) {
...
+
+	// Determine if an add or delete
+	action := r.URL.Query().Get("action")
+	if action == "add" {
+		updateOperator = "$addToSet"
+	} else if action == "delete" {
+		updateOperator = "$pull"
+	} else {
+		log.Printf("ERROR: Incorrect action to edit user favorites.")
+		w.WriteHeader(http.StatusBadRequest)
```
Comment (kftang) [inline routes/users.go:185]: return, but i think this is redundant from the above comment?

**Review by kftang — DISMISSED** [review]

File: routes/users.go (hunk 2 above)
Comment (janinebar) [reply routes/users.go:179]: hmmmm well it works and andrea says that route is good

Comment (janinebar) [reply routes/users.go:179]: so i think ill keep it

File: routes/users.go (hunk 3 above)
Comment (janinebar) [reply routes/users.go]: ok yeah sorry keep forgetting this

File: routes/users.go (hunk 5 above)
Comment (janinebar) [reply routes/users.go:185]: done

File: routes/users.go (hunk 4 above)
Comment (janinebar) [reply routes/users.go:194]: done

File: server.go (hunk 1 above)
Comment (janinebar) [reply server.go]: fixed

**Review by ngynandrea — APPROVED** [review]

**Review by yasirayounus — APPROVED** [review]

**Pull request merged**